common/src/main/java/tachyon/heartbeat/HeartbeatScheduler.java

```diff
@@ -63,19 +66,23 @@ public static void addTimer(ScheduledTimer timer) {
   public static synchronized void removeTimer(ScheduledTimer timer) {
     Preconditions.checkNotNull(timer);
     sLock.lock();
-    sTimers.remove(timer.getThreadName());
-    sLock.unlock();
+    try {
+      sTimers.remove(timer.getThreadName());
+    } finally {
+      sLock.unlock();
+    }
```

Comment:
LGTM. I think functionality-wise, having `try...catch...` block is strictly better.